### ISSUE TITLE
There are some duplicate implementations of (Clock, MemoryReactor) combiner

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6381]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6381|
|Type|defect|
|Created|2013-03-19 05:17:25Z|
|Branch|https://github.com/twisted/twisted/tree/clockmemoryreactor-6381|

We have a bunch of subclasses of Clock and MemoryReactor in various places.  We should remove the duplication.

<details><summary>Searchable metadata</summary>

```
trac-id__6381 6381
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_clockmemoryreactor_6381 branches/clockmemoryreactor-6381
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1363670245000000 1363670245000000
changetime__1363711046000000 1363711046000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [37659]) Branching to 'clockmemoryreactor-6381'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Pretty simple.  Here are the [build results](https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/clockmemoryreactor-6381).  pyflakes results seem a bit nonsensical, not sure what's going on there.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

And by pyflakes I meant twistedchecker, I guess.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Wonderful, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [37671]) Merge clockmemoryreactor-6381

Author: exarkun
Reviewer: therve
Fixes: [#6381](https://github.com/twisted/twisted/issues/6381)

Eliminate the various duplicate versions of a `Clock` / `MemoryReactor` subclass
used in a number of different tests.
